misaligned.cpp

```diff
@@ -1,21 +1,32 @@
-#include <iostream>
 #include <assert.h>
+#include <iostream>
 
-int printColorMap() {
+void checkPairNumberValidity(int pair_number)
+{
+    assert(pair_number ==0 );
+    assert(pair_number >= 1 && pair_number <= 25);
+}
+
+void printColorMap()
+{
     const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
     const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
-    int i = 0, j = 0;
-    for(i = 0; i < 5; i++) {
-        for(j = 0; j < 5; j++) {
-            std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
+    int         i = 0, j = 0;
+    int pair_number;
+    for (i = 0; i < 5; i++)
+    {
+        for (j = 0; j < 5; j++)
+        {
+            pair_number = (i * 5 + j)+1;
+            std::cout << pair_number << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
+            checkPairNumberValidity(pair_number);
         }
     }
-    return i * j;
 }
 
-int main() {
-    int result = printColorMap();
-    assert(result == 25);
+int main()
+{
+    printColorMap();

```

Comment:
the production code is still buggy. This is exactly what we need to fix by having strong test cases. the output should be verifiable by test code.